tests/etc/better-addresses.ini

```diff
@@ -67,6 +67,33 @@ targets = {
    }
 
 
+[config:pushsafer]
+; https://www.pushsafer.com/en/pushapi
+; https://www.pushsafer.com/en/pushapi_ext
+targets = {
+    'nagios': {
+        'private_key': '3SAz1a2iTYsh19eXIMiO',
+        'device': '52|65|78',
+        'icon': 64,
+        'sound': 2,
+        'vibration': 'blank',  # ?
```

Comment:
Dear @appzer,

may I humbly ask you to review this line?
```
'vibration': 'blank',  # ?
```

The documentation [^1] says about the value range:
> `blank`=device default or a number `1`-`3`.

Does it actually mean that the client should send `&v=` in order to enable the vibration feature with device defaults, without any value?

With kind regards,
Andreas.

[^1]: https://www.pushsafer.com/en/pushapi_ext#API-V

Reply:
> &v=

that correct! BLANK means without value = device default!

Reply:
Thank you. I will take care about proper serialization of that parameter, which will have to be handled specially, because `None` values would otherwise not be propagated.

May I ask if there are **any other parameters like this**, where including them with an empty value into the query string actually signals something?

Reply:
b021547a75d3 now implements the special handling for the `vibration` parameter, _to make sure it is submitted empty_, while all the other parameters will be omitted when empty.

In order to use the device-default vibration, mqttwarn users will have to explicitly configure `vibration=0` now. I think this slight breaking change is acceptable while fixing an actual bug, that previously, vibrations have been turned on for each notification.

-- https://www.pushsafer.com/en/pushapi_ext#API-V

Reply:
Hi, 

yes, the parameter sound can be empty for device default sound.
0 = silent
1-62 plays a pushsafer sound

I cant change the API handling.

Kind Regards Kevin

Reply:
you also can use the special handling for the sound parameter!

Thanks Kevin

Reply:
> The parameter `sound` can be empty for device-default sound.

I see. Thanks for the clarification.

> I cant change the API handling.

I understand.

Reply:
Regarding the _Sound_ parameter, the documentation says:
> If the parameter is not passed or left blank, the device plays the default sound.

I think in this case it is fine, because the default behavior actually is to play the default sound _when the parameter is omitted_.

Maybe we have a misunderstanding here: Would it actually have been fine to omit the `vibration` parameter as well?

-- https://www.pushsafer.com/en/pushapi_ext#API-S

Reply:
Maybe we have a misunderstanding here: Would it actually have been fine to omit the vibration parameter as well? Oh well, I am right now re-reading the documentation, and it clearly states:

> **How often** the device should vibrate when receiving a push-notification.

I overlooked that detail. Apologies. I guess I will revert all my changes about this then. I misinterpreted the parameter to be a able to turn vibrations off.

-- https://www.pushsafer.com/en/pushapi_ext#API-V

Reply:
> Does it actually mean that the client should send `&v=` in order to enable the vibration feature with device defaults, without any value?

This was my initial question. Can you confirm it does not make any difference if the client sends `&v=`, or just omit it?

Reply:
6e68f7faa removes the special handling again, if you agree. Sorry again for the noise.

Reply:
> This was my initial question. Can you confirm it does not make any difference if the client sends `&v=`, or just omit it?

that makes no difference